src/python/CMSSpark/data_aggregation.py

```diff
@@ -0,0 +1,385 @@
+             "JOIN ddf ON fdf.f_dataset_id = ddf.d_dataset_id "\
+             "JOIN phedex_df ON bdf.b_block_name = phedex_df.block_name") % ','.join(cols)
+
+    if verbose:
+        print 'Will run query to generate temp file_block_site table'
+
+    result = run_query(query, sql_context, verbose)
+    result.registerTempTable('f_b_all_df')
+
+    query_distinct = ("SELECT DISTINCT * FROM f_b_all_df ORDER  BY file_name")
+    result_distinct = run_query(query_distinct, sql_context, verbose)
+    result_distinct.registerTempTable('f_b_s_df')
+
+    if verbose:
+        print 'Temp table from joined DDF, FDF, BDF and PhEDEx'
+        print 'After DISTINCT: ' + str(result.count()) + ' -> ' + str(result_distinct.count())
+        result_distinct.show(20)
+        print_rows(result_distinct, query_distinct, verbose, 5)
+        result_distinct.printSchema()
+        print 'Finished file_block_site generation'
+
+
+def clean_site_name(s):
+    split = s.split('_')
+    split = split[0:3]
+
+    # Remove empty strings which may appear when s is T0_USA_
+    split = filter(None, split)
+
+    join = '_'.join(split)
+    return join
+
+
+def main():
+    "Main function"
+    optmgr = OptionParser()
+    opts = optmgr.parser.parse_args()
+
+    print("Input arguments: %s" % opts)
+
+    start_time = time.time()
+    verbose = opts.verbose
+    yarn = opts.yarn
+    inst = opts.inst
+    date = opts.date
+    fout = opts.fout
+
+    if  inst.lower() in ['global', 'phys01', 'phys02', 'phys03']:
+        inst = inst.upper()
+    else:
+        raise Exception('Unsupported DBS instance "%s"' % inst)
+
+    # Create spark context
+    ctx = spark_context('cms', yarn, verbose)
+
+    quiet_logs(ctx)
+
+    # Create SQL context to be used for SQL queries
+    sql_context = HiveContext(ctx)
+
+    # Initialize DBS tables
+    dbs_tables(sql_context, inst=inst, verbose=verbose)
+
+    # Initialize PhEDEx table to be used in file_block_site table
+    phedex_tables(sql_context, verbose=verbose)
+
+    # Register clean_site_name to be used with SQL queries
+    sql_context.udf.register("clean_site_name", clean_site_name)
+
+    # Create temp table with file name, block name, site name and site from PhEDEx
+    create_file_block_site_table(ctx, sql_context, verbose)
+
+    cmssw_start_time = time.time()
+    aggregated_cmssw_df = run_agg_cmssw(date, fout, ctx, sql_context, verbose)
+    cmssw_elapsed_time = elapsed_time(cmssw_start_time)
+
+    aaa_start_time = time.time()
+    aggregated_aaa_df = run_agg_aaa(date, fout, ctx, sql_context, verbose)
+    aaa_elapsed_time = elapsed_time(aaa_start_time)
+
+    eos_start_time = time.time()
+    aggregated_eos_df = run_agg_eos(date, fout, ctx, sql_context, verbose)
+    eos_elapsed_time = elapsed_time(eos_start_time)
+
+    jm_start_time = time.time()
+    aggregated_jm_df = run_agg_jm(date, fout, ctx, sql_context, verbose)
+    jm_elapsed_time = elapsed_time(jm_start_time)
+
```

Comment:
you may consider to use persist(StorageLevel.MEMORY_AND_DISK) for all created dataframes here since you're going to reuse them later.